packages/suite/src/views/onboarding/steps/Final/index.tsx

```diff
@@ -232,27 +266,24 @@ export const FinalStep = () => {
                     {state === 'rename' && (
                         <SetupActions>
                             <Input
+                                width={250}
                                 noTopLabel
-                                noError
                                 value={label}
+                                bottomText={error}
                                 placeholder={DEFAULT_LABEL}
-                                inputState={label.length > MAX_LABEL_LENGTH ? 'error' : undefined}
-                                onChange={(event: React.FormEvent<HTMLInputElement>) =>
-                                    setLabel(event.currentTarget.value)
-                                }
+                                inputState={error ? 'error' : undefined}
+                                onChange={handleChange}
                                 data-test="@settings/device/label-input"
                             />
-                            <Button
+                            <RenameButton
                                 onClick={async () => {
                                     await onRename();
```

Comment:
we miss analytics here ☹️ is it possible to reuse DeviceLabel component here somehow? instead of the duplication